interface/src/lib.rs

```diff
@@ -337,11 +331,11 @@ pub fn calculate_address(
 ) -> Address {
   // calculate address by hashing the template, blob, principal, and nonce
   let mut hasher = Hasher::new();
-  hasher.update(template);
+  hasher.update(&template.0);
```

Comment:
wouldn't an Into method for a byte slice make this more readable?

Reply:
`Into` consumes the value.  I will add implementations of `AsRef` instead.